File: src/features/waste/service/wasteService.js
import { createCrudService } from '../../../shared/service/createCrudService';
import { createWasteCategory } from '../domain/WasteCategory';
import { wasteRepository } from '../infrastructure/wasteRepository';

export const wasteService = createCrudService({
  repository: wasteRepository,
  createEntity: createWasteCategory,
  idField: 'categoryId',
});

File: backend/src/main/java/com/oose/labsafety/waste/service/WasteService.java
package com.oose.labsafety.waste.service;

import com.oose.labsafety.common.service.CrudService;
import com.oose.labsafety.waste.domain.WasteCategory;
import com.oose.labsafety.waste.infrastructure.WasteRepository;
import org.springframework.stereotype.Service;

@Service
public class WasteService extends CrudService<WasteCategory> {
    public WasteService(WasteRepository repository) {
        super(repository);
    }
}
